records_mover/url/s3/s3_directory_url.py

```diff
@@ -3,6 +3,148 @@
+
+
+def _wait_for_transfer_job(project_id: str, job_name: str) -> None:
+    # TODO: Pass in correct creds
+    storagetransfer = googleapiclient.discovery.build('storagetransfer', 'v1')
+
+    # https://github.com/karthikey-surineni/bigquery-copy-dataset/blob/fb6be5aee9898cba79387a2f95fb54b8b1119a92/src/storage/storage_transfer.py
+    filterString = (
+        '{{"project_id": "{project_id}", '
+        '"job_names": ["{job_name}"]}}'
+    ).format(project_id=project_id, job_name=job_name)
+
+    done = False
+    logger.info("Awaiting completion of Google Storage Transfer Service job...")
+    while not done:
+        result = storagetransfer.transferOperations().list(
+            name="transferOperations",
+            filter=filterString).execute()
+        print('Result of transferOperations/list: {}'.format(
+            json.dumps(result, indent=4, sort_keys=True)))
+        if result != {}:
+            done = result['operations'][0]['metadata']['status'] != 'IN_PROGRESS'
+        time.sleep(10)
+    logger.info("Google Storage Transfer Service job complete.")
+
+
+
+def copy_via_gcp_data_transfer(loc: BaseDirectoryUrl,
+                               other_loc: BaseDirectoryUrl) -> None:
+    # https://cloud.google.com/storage-transfer/docs/create-manage-transfer-program#python
+    storagetransfer = googleapiclient.discovery.build('storagetransfer', 'v1')
+    # TODO
+    description = "records-mover one-time job"
+    # TODO
+    project_id = "bluelabs-tools-dev"
+    # {
+    #   "description": "bluelabs-scratch-bluelabs-test-recordsmover_once",
+    #   "projectId": "bluelabs-tools-dev",
+    #   "transferSpec": {
+    #     "gcsDataSink": {
+    #       "bucketName": "bluelabs-test-recordsmover"
+    #     },
+    #     "awsS3DataSource": {
+    #       "bucketName": "bluelabs-scratch",
+    #       "awsAccessKey": {
+    #         "accessKeyId": "AKIAJ5RGGUDLXXMDAYHQ",
+    #         "secretAccessKey": "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"
+    #       }
+    #     },
+    #     "objectConditions": {
+    #       "includePrefixes": [
+    #         "vince.broz"
+    #       ]
+    #     },
+    #     "transferOptions": {
+    #       "overwriteObjectsAlreadyExistingInSink": false,
+    #       "deleteObjectsUniqueInSink": false,
+    #       "deleteObjectsFromSourceAfterTransfer": false
+    #     }
+    #   },
+    #   "schedule": {
+    #     "scheduleStartDate": {
+    #       "year": 2020,
+    #       "month": 11,
+    #       "day": 10
+    #     },
+    #     "scheduleEndDate": {
+    #       "year": 2020,
+    #       "month": 11,
+    #       "day": 10
+    #     }
+    #   },
+    #   "status": "ENABLED"
+    # }
+    # TODO
+    current_day = 10
+    # TODO
+    current_month = 11
+    # TODO
+    current_year = 2020
+    # TODO
+    sink_bucket = "bluelabs-test-recordsmover"
+    # TODO
+    source_bucket = "bluelabs-scratch"
+    # TODO
+    access_key_id = loc.s3_creds.access_key_id
+    # TODO
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker